lib/src/base/types/Dynamic.vue

```diff
@@ -101,6 +158,45 @@ export default {
 
           this.$emit("load", false);
         });
+    },
+    addStatic() {
+      this.instanceNameError = false
+      if (this.globalSectionMode && this.instanceName === '') {
+        this.instanceNameError = true
+        return
+      }
+      this.$emit("addSectionType", {
+        name: this.props.name,
+        type: 'dynamic',
+        id: this.id,
+        weight: this.weight,
+        auto_insertion: this.autoInsert,
+        instance_name: this.instanceName
+      });
+    },
+    getGlobalType() {
+      const token = this.$cookies.get("sections-auth-token");
+      const config = {
+        headers: sectionHeader({ token }),
+      };
+      const URL =
+          `${this.$sections.serverUrl}/project/${this.$sections.projectId}/global-instances/${this.props.linked_to}`;
+      this.$emit("load", true);
+
+      this.$axios.get(URL, config).then((res) => {
+        if (res && res.data) {
+          this.$emit("load", false);
+          this.autoInsert = res.data.auto_insertion
+          if (res.data.pages && res.data.pages.length > 0) {
+            this.pages = res.data.pages.map(p => p.path)
+          }
+          this.instanceName = res.data.name
+        }
+      })
+          .catch((error) => {
+            this.$emit("load", false);
+            this.showToast("Error", "error", error.response.data.message, error.response.data.options);
+          });
```

Comment:
this is the EXACT same code as the one in the configurable, local and static components... why don't you put it in a centralized file and call it from these four places ?